app/shared/design-system/components/primitives/numericGestureControl.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @description
+ * - detects pointer movements in the specified direction
+ * - avails an appropriate cursor
+ * - dispatches {onValueChange} method while pointermove and pointerdown are active
+ * - dispatched {onValueChange} recieves a new value: old value + accumulation of the pointer move axis
+ * @example
+ * numericGestureControl(document.querySelector('input'), {
+ *   onValueChange: (event) => {
+ *     event.preventDefault();
+ *     event.target.value = event.value;
+ *     event.target.select();
+ *   }
+ * });
+ */
+export const numericGestureControl = (
+  targetNode,
+  { initialValue = 0, direction = "horizontal", onValueChange = ({}) => null }
+) => {
+  const { ownerDocument } = targetNode;
+  const eventNames = ["pointerup", "pointerdown", "pointermove"];
+  const state = {
+    value: initialValue,
+    cursor: cursorNode.cloneNode(true),
+    offset: 0,
+    velocity: direction === "horizontal" ? 1 : -1,
+    position: direction === "horizontal" ? "left" : "top",
+    rotation: direction === "horizontal" ? "rotate(0deg)" : "rotate(90deg)",
+    direction: direction === "horizontal" ? "ew-resize" : "ns-resize",
+    dropShadow: `drop-shadow(${
+      direction === "horizontal" ? "0 1px" : "1px 0"
+    } 1.1px rgba(0,0,0,.4))`,
```

Comment:
IMO this the UI concern and should not be mixed with the gesture state itself. That being said, I think we should still calculate and return the styles from the hook but I would scope them under a separate object like `css` or `styles` to make it 1. clear there the separation is and 2. easier to get final styles using our design-system.

Reply:
agree